### PR TITLE
Improve form choice providers

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR updates naming of form choice providers and moves them to the same namespace.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Localization & Geolocation pages should not break, as they are the only ones that uses form choice providers at the moment.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9205)

### CODE REVIEW

**Comment by sarjon** [comment]
@mickaelandrieu maybe these form choice providers should be documented somewhere as they can be reused a lot?

File: src/Core/Form/ChoiceProvider/CountryByIdChoiceProvider.php [hunk 1]
```diff
@@ -24,35 +24,35 @@
...
+    private $langId;
 
     /**
-     * @param LegacyContext $legacyContext
+     * @param int $langId
      * @param CountryDataProvider $countryDataProvider
      */
     public function __construct(
-        LegacyContext $legacyContext,
+        $langId,
```
Comment (mickaelandrieu) [inline src/Core/Form/ChoiceProvider/CountryByIdChoiceProvider.php:52]: :+1:

**Comment by mickaelandrieu** [comment]
@sarjon you're right, but we need to think about a common use case a PrestaShop developer could have to make it a cookbook in docs: do you have something in mind?

**Comment by sarjon** [comment]
what do you mean by 'common use case'?

**Comment by mickaelandrieu** [comment]
Some tasks where PrestaShop developers must use form choice provider

**Comment by sarjon** [comment]
i think this will be most useful when module developers will be able to create modern controllers and use choice providers it in their forms.

**Review by mickaelandrieu — APPROVED** [review]

**Review by mickaelandrieu — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Thank you @sarjon

**Comment by mickaelandrieu** [comment]
Thanks @sarjon  and @ntiepresta for the review

**Pull request merged**